### PR TITLE
Added context menu to hide sidebar sections

### LINKED ISSUE #8233: Context menu on the sidebar
### What's the Problem?

It is possible to hide sections from the sidebar using the section's contextual menu.
It is not possible to view a section without opening the settings menu.

### Solution/Idea

We can add a context menu on the sidebar (there is none yet) to show/hide all sections.

### Alternatives

None

### Priorities

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish W | Must |
| This proposal will allow end users to accomplish X | Should |
| This proposal will allow developers to accomplish Y | Could |
| This proposal will allow end users to accomplish Z | Won't |

### Files Version

_No response_

### Windows Version

_No response_

### Comments

_No response_

### PR DESCRIPTION
**Pull request by oleklukasiewicz**
**Resolved / Related Issues**
- Closes #8233 

**Details of Changes**
- Added context menu to sidebar
- Added 3 new localizations in en-US

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
image

### CODE REVIEW

**Comment by gave92** [comment]
Nice, with this the "Sidebar" section in settings could potentially be removed. @yair100?

**Comment by oleklukasiewicz** [comment]
> Nice, with this the "Sidebar" section in settings could potentially be removed. @yair100?

If the sections were removed from the settings it would make it easier to centralize the sidebar settings. Now, the sidebar options are split into several view models (AppearanceViewModel, SidebarViewModel and other), which is why the code is written in C# and not in XAML with the proper data bindings.

**Comment by yair100** [comment]
> Nice, with this the "Sidebar" section in settings could potentially be removed. @yair100?

@gave92 there is still a benifit of having the settings available in the dialog.

**Comment by yaichenbaum** [comment]
@aleksanderlukasiewicz is there a scenario where someone will want to hide all the sections?

**Review by yair100 — CHANGES_REQUESTED** [review]

**Comment by gave92** [comment]
> @aleksanderlukasiewicz is there a scenario where someone will want to hide all the sections?

If removing all sections would make the sidebar completely disappear that yes

**Comment by yaichenbaum** [comment]
@gave92 that's not currently part of the spec

File: src/Files/UserControls/SidebarControl.xaml.cs [hunk 1]
```diff
@@ -332,6 +405,89 @@ private async void Sidebar_PointerPressed(object sender, PointerRoutedEventArgs
             }
         }
 
+        private void PaneRoot_RightTapped(object sender, RightTappedRoutedEventArgs e)
+        {
+            var sidebarContextMenu = new MenuFlyout();
```
[the remark this answers is not in the document]
Comment (aleksanderlukasiewicz) [reply src/Files/UserControls/SidebarControl.xaml.cs:410]: Sure

[the remark this answers is not in the document]
Comment (aleksanderlukasiewicz) [reply src/Files/UserControls/SidebarControl.xaml.cs]: Done!

**Comment by oleklukasiewicz** [comment]
> @aleksanderlukasiewicz is there a scenario where someone will want to hide all the sections?

@yair100  Yea there is an option for this.

**Comment by yaichenbaum** [comment]
@aleksanderlukasiewicz my question was if the option is necessary, can you remove it?

**Comment by oleklukasiewicz** [comment]
@yair100  Okay sure!

**Comment by oleklukasiewicz** [comment]
@yair100  Something like this?
image

**Comment by yaichenbaum** [comment]
@aleksanderlukasiewicz yes

**Comment by oleklukasiewicz** [comment]
@yair100 Done!

**Comment by yaichenbaum** [comment]
@aleksanderlukasiewicz hiding the network section and then showing it again appears to collapse the other sections.

**Comment by oleklukasiewicz** [comment]
@yair100 ok I'll look into it

**Comment by oleklukasiewicz** [comment]
@yair100 This looks like a much deeper problem related to more than just the network section. I noticed similar behavior when I changed section visibility from the settings page.

**Comment by yair100** [comment]
> @yair100 This looks like a much deeper problem related to more than just the network section. I noticed similar behavior when I changed section visibility from the settings page.

For some reason it only appears to happen from the context menu on my end 🤔

**Comment by oleklukasiewicz** [comment]
> > @yair100 This looks like a much deeper problem related to more than just the network section. I noticed similar behavior when I changed section visibility from the settings page.
> 
> For some reason it only appears to happen from the context menu on my end 🤔

Strange, I get this error even in version 2.1.7.0 without the context menu. 🤨
I think it's a problem (at least from what I have) with the sidebar control itself.

The menu uses exactly the same mechanism as the settings so it should have the same effect.

**Review by gave92 — APPROVED** [review]

**Review by yair100 — APPROVED** [review]

**Pull request merged**